crates/core/tedge_agent/src/http_rest.rs

```diff
@@ -0,0 +1,299 @@
+        let value = config.query(TmpPathSetting).unwrap();  // TODO
+        let value: PathBuf = value.into();
+        let root_path = PathBuf::from(format!("{}/{WHAT_TO_NAME_THIS}", value.display()));
+        root_path
+    };
+}
+
+#[cfg(test)]
+lazy_static::lazy_static! {
+    static ref FILE_TRANSFER_ROOT_PATH: PathBuf = {
+        let ttd = tedge_test_utils::fs::TempTedgeDir::new();
+        dbg!(&ttd.path());
+        let root_path = PathBuf::from(format!("{}/{WHAT_TO_NAME_THIS}", ttd.path().display()));
+        root_path
+    };
+}
+
+fn separate_path_and_file_name(input: &str) -> Option<(PathBuf, &str)> {
+    let (mut relative_path, file_name) = input.rsplit_once("/")?;
+
+    if relative_path.starts_with("/") {
+        relative_path = &relative_path[1..];
+    }
+    let relative_path = PathBuf::from(relative_path);
+    Some((relative_path, file_name))
+}
+
+async fn put(mut request: Request<Body>) -> Result<Response<Body>, AgentError> {
+    let uri = request.uri().to_string();
+    let mut response = Response::new(Body::empty());
+    if let Some((relative_path, file_name)) = separate_path_and_file_name(&uri) {
+        let root_path = &FILE_TRANSFER_ROOT_PATH;
+        let directories_path = root_path.join(relative_path);
+
+        if let Err(_err) = create_directories(&directories_path) {
+            *response.status_mut() = hyper::StatusCode::FORBIDDEN;
+        }
+
+        let full_path = directories_path.join(file_name);
+
+        match stream_request_body_to_path(&full_path, request.body_mut()).await {
+            Ok(()) => {
+                *response.status_mut() = hyper::StatusCode::CREATED;
+            }
+            Err(_err) => {
+                *response.status_mut() = hyper::StatusCode::FORBIDDEN;
+            }
+        }
+    } else {
+        *response.status_mut() = hyper::StatusCode::FORBIDDEN;
+    }
+    Ok(response)
+}
+
+async fn get(request: Request<Body>) -> Result<Response<Body>, AgentError> {
+    let root_path = &FILE_TRANSFER_ROOT_PATH;
+    let full_path = PathBuf::from(format!("{}{}", root_path.display(), request.uri()));
+    dbg!(&full_path);
+
+    //dbg!(full_path.exists(), full_path.is_dir());
+    //if full_path.exists() || full_path.is_dir() {
+    //    let mut response = Response::new(Body::empty());
+    //    *response.status_mut() = hyper::StatusCode::NOT_FOUND;
+    //    return Ok(response);
+    //}
+
+    let mut file = tokio::fs::File::open(full_path).await?;
+
+    let mut contents = vec![];
+    file.read_to_end(&mut contents).await?;
+
+    let output = String::from_utf8(contents)?;
+
+    Ok(Response::new(Body::from(output)))
+}
+
+async fn delete(request: Request<Body>) -> Result<Response<Body>, AgentError> {
+    let root_path = &FILE_TRANSFER_ROOT_PATH;
+    let full_path = PathBuf::from(format!("{}{}", root_path.display(), request.uri()));
+
+    let mut response = Response::new(Body::empty());
+
+    match tokio::fs::remove_file(&full_path).await {
```

Comment:
This looks equally dangerous. Removing `/bin/init` would be possible with this, as far as I can see.